### PR TITLE
kata-deploy: Ensure we cover SHIMS / DEFAULT_SHIM as part of our tests

### PR DESCRIPTION
**Pull request by fidencio**
---

k8s: tests: Take advantage of `SHIMS` and `DEFAULT_SHIM` env vars

We don't have to do any sed to replace the runtimeclass being used by
the moment we start taking advantage of the `DEFAULT_SHIM` environment
variable exposed merged in the previous commits.

---

tests: k8s: Ensure the runtime classes are properly created

With these 2 simple checks we can ensure that we do not regress on the
behaviour of allowing the runtime classes / default runtime class to be
created by the kata-deploy payload.

---

### CODE REVIEW

**Comment by stevenhorsman** [comment]
So just to double-check I'm understanding correctly, as I don't think I saw the `CREATE_DEFAULT_RUNTIMECLASS` get added - is the idea that if that is set to true then kata-deploy will create a runtimeClass of `kata` that points to the default shim?

Then this update allows us to avoid overwriting the runtimeClassName in the test yamls?

File: tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2018 Intel Corporation
```
Comment (stevenhorsman) [inline tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats:3]: nit: Copyright year

**Comment by fidencio** [comment]
> So just to double-check I'm understanding correctly, as I don't think I saw the `CREATE_DEFAULT_RUNTIMECLASS` get added - is the idea that if that is set to true then kata-deploy will create a runtimeClass of `kata` that points to the default shim?
> 
> Then this update allows us to use this to avoid overwriting the runtimeClassName in the test yamls?

You got it correctly, 100% correctly.

File: tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+setup() {
+	expected_runtime_classes=2
+}
+
+@test "Test runtimeclasses are being properly created" {
+	echo "FIDENCIO | DEBUG"
+	current_runtime_classes=$(kubectl get runtimeclasses | grep "kata" | wc -l)
+	echo "FIDENCIO | current_runtime_classes: $current_runtime_classes"	
+	echo "FIDENCIO | expected_runtime_classes: $expected_runtime_classes"
```
Comment (stevenhorsman) [inline tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats:19]: I think you might have to force the test to fail to get BATS to show the logs? (or alter a setting elsewhere for that for the whole run?)

File: tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	expected_runtime_classes=2
```
Comment (stevenhorsman) [inline tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats:12]: Just to double check - the reason we have two expected runtime classes are the one we are testing and `kata` which defaults to map to it?

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

File: tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats (hunk 3 above)
Comment (fidencio) [reply tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats:12]: Exaclty!  Do you want me to add a comment making it explicit? (I do think it's a good idea)

Comment (fidencio) [reply tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats:16]: I did.

Comment (stevenhorsman) [reply tests/integration/kubernetes/kata-deploy-ensure-runtimec-classes-created.bats:16]: Thanks - that might have been what I was hinting at ;)

**Review by zvonkok — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**